File: backend/src/main/java/com/tiago/crud/service/UserService.java
package com.tiago.crud.service;

import com.tiago.crud.model.User;
import com.tiago.crud.repository.UserRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class UserService {

    private final UserRepository repository;

    public UserService(UserRepository repository) {
        this.repository = repository;
    }

    public List<User> getUsers() {
        return repository.findAll();
    }

    public User createUser(User user) {
        return repository.save(user);
    }

    public User updateUser(Long id, User user) {

        User existingUser = repository.findById(id)
                .orElseThrow(() -> new RuntimeException("User not found"));

        existingUser.setName(user.getName());
        existingUser.setEmail(user.getEmail());

        return repository.save(existingUser);
    }

    public void deleteUser(Long id) {
        repository.deleteById(id);
    }
}

File: frontend/src/services/userService.js
import axios from "axios"

const API_URL = "http://localhost:8080/users"

export default {

    getUsers() {
        return axios.get(API_URL)
    },

    createUser(user) {
        return axios.post(API_URL, user)
    },

    updateUser(id, user) {
        return axios.put(`${API_URL}/${id}`, user)
    },

    deleteUser(id) {
        return axios.delete(`${API_URL}/${id}`)
    }

}